### PR TITLE
client-sdk/go/crypto: Implement secp256k1 signer

### LINKED ISSUE #95: Implement missing secp256k1 signer
Currently the secp256k1 signer implementation in the [Go client SDK](https://github.com/oasisprotocol/oasis-sdk/tree/main/client-sdk/go/crypto/signature/secp256k1) is missing and should be filled in. We could use the `github.com/btcsuite/btcd/btcec` library (unless there are any better alternatives).

### PR DESCRIPTION
**Pull request by abukosek**
Closes #95.

### CODE REVIEW

**Review by kostko — COMMENTED** [review]
Would be cool to have an E2E test where we use a secp256k1 account to sign a transaction. This means you would want to add a test key or three based on secp256k1.

**Comment by abukosek** [comment]
OK, I'll add that :)

**Comment by pro-wh** [comment]
do we have any external things to test compatibility against? I have a draft of similar signing/verifying in the typescript lib, so if you have something, I'd like to test the typescript part against that as well.

**Comment by abukosek** [comment]
The library I'm using (`btcec`) already has extensive tests, so I didn't feel the need to repeat them here, since this PR basically just wraps their library.  I do test if our code properly generates a signature that's in their examples, but that's just a quick sanity check.
If you find any offical test vectors, we can add those in a separate PR, just to be extra sure.

File: client-sdk/go/testing/testing.go [hunk 1]
```diff
@@ -22,11 +25,22 @@ func newTestKey(seed string) TestKey {
 	}
 }
 
+func newSecp256k1TestKey(seed string) TestKey {
```
Comment (kostko) [inline client-sdk/go/testing/testing.go:28]: Rename the other one to `newEd25519TestKey`?

File: runtime-sdk/Cargo.toml [hunk 2]
```diff
@@ -13,6 +13,7 @@ oasis-runtime-sdk-macros = { path = "../runtime-sdk-macros", optional = true }
 serde = { version = "1.0.118", features = ["derive"] }
 serde_bytes = "0.11.5"
 ed25519-dalek = { version = "1.0.1", features = ["serde"] }
+sha2 = "0.9.3"
 k256 = { version = "0.6.0", features = ["ecdsa", "sha256"] }
```
Comment (kostko) [inline runtime-sdk/Cargo.toml:17]: Remove the `sha256` feature now?

File: runtime-sdk/src/testing/keys.rs [hunk 3]
```diff
@@ -36,6 +36,41 @@ macro_rules! test_key {
     }
 }
 
+/// Define a secp256k1 test key.
+macro_rules! test_key_secp256k1 {
```
Comment (kostko) [inline runtime-sdk/src/testing/keys.rs:40]: Rename the other one to `test_key_ed25519`.

**Review by kostko — APPROVED** [review]

**Comment by abukosek** [comment]
Thanks!

**Pull request merged**